File: lib/folavirt/networking/Query.py
#!/usr/bin/env python2
# -*- coding: utf-8 -*-

import json

class Query():
    """
    Zapytanie
    
    @author: michal
    @group folavirt.networking: 
    """
    def __init__(self, command = "", array = {}):
        self.command = command
        self.array = array
    
    command = ""
    array = {}
    
    def setCommand(self, command):
        """
        Ustawia polecenie
        
        @param Polecenie
        """
        self.command = command
    
    def getCommand(self):
        """
        Zwraca polecenie
        
        @param void
        @return Polecenie
        """
        return self.command
    
    def setData(self, array):
        self.array = array
      
    def getData(self):
        return self.array
        
    def getJSON(self):
        retarray = {}
        retarray['command'] = self.command
        retarray['data'] = self.array
        
        return json.dumps(retarray)

    def setFromJSON(self, jsondata):
        retarray = json.loads(jsondata)
        
        self.command = retarray['command']
        self.array = retarray['data']

File: www/module/Application/src/Application/Model/Folavirt/Networking/Query.php
<?php

namespace Application\Model\Folavirt\Networking;

/**
 * Zapytanie do agenta folavirt
 * 
 * @package Model\Folavirt
 * @see lib/folavirt/networking/Query
 */
class Query
{
	public $command;
	public $data = array();
	
	/**
	 * Ustawia polecenie
	 * 
	 * @param string
	 * @return void
	 */
	public function setCommand($command)
	{
		$this -> command = $command;
	}
	
	/**
	 * Zwraca polecenie
	 * 
	 * @param void
	 * @return string
	 */
	public function getCommand()
	{
		return $this -> command;
	}
	
	/**
	 * Ustawia parametry
	 * 
	 * @param array
	 * @return void
	 */
	public function setData($data)
	{
		$this -> data = $data;
	}
	
	/**
	 * Zwraca parametry
	 * 
	 * @param void
	 * @return array
	 */
	public function getData()
	{
		return $this -> data;
	}
	
	/**
	 * Zwraca zapytania w formacie JSON
	 * 
	 * @param void
	 * @return json
	 */
	public function getJSON()
	{
		$retarray = array();
		$retarray['command'] = $this -> command;
		$retarray['data'] = $this -> data;
		
		return json_encode($retarray);
	}
	
	
}
